### PR TITLE
Remove Platform from DialogueConnector

### LINKED ISSUE #220: Remove Platform from DialogueConnector
A platform should internally take care of instantiating Agents and Users, and connecting them via DialogueConnector.
 * Implement a method platform.connect(user_id)
    - if user_id is given then it’s a returning user (w/ history)
 * Implement platform.disconnect(), message(), feedback()

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #220

### CODE REVIEW

File: dialoguekit/platforms/platform.py [hunk 1]
```diff
@@ -44,13 +44,79 @@ def display_agent_utterance(self, utterance: Utterance) -> None:
...
+    def feedback(self, user_id: str, utterance_id: str, value: int) -> None:
+        """Gets called every time there is a new utterance feedback.
+
+        Args:
+            user_id: User ID.
+            utterance_id: Utterance ID.
+            value: Feedback value.
+        """
+        # TODO implement feedback
+        # Issue: https://github.com/iai-group/DialogueKit/issues/219
```
Comment (NoB0) [inline dialoguekit/platforms/platform.py:122]:
```suggestion
        # Issue: https://github.com/iai-group/DialogueKit/issues/219
        pass
```

File: dialoguekit/platforms/terminal_platform.py [hunk 2]
```diff
@@ -3,37 +3,48 @@
...
-
-    def listen_for_user_input(self) -> None:
-        """Listens for the user input.
+    def __init__(self, agent_class: Type[Agent], user_id: str = None) -> None:
+        """Represents a terminal platform. It handles a single user.
 
         Args:
-            callback: Function to call on user input
+            agent_class: The class of the agent.
+            user_id: User ID (default: terminal_user).
```
Comment (NoB0) [inline dialoguekit/platforms/terminal_platform.py:20]:
```suggestion
            user_id: User ID. Defaults to terminal_user.
```
To follow the format used in the rest of the code.

File: dialoguekit/platforms/terminal_platform.py [hunk 3]
```diff
@@ -3,37 +3,48 @@
...
 
 
 class TerminalPlatform(Platform):
-    def __init__(self):
-        """Represents a platform."""
-        self._user_callback: Callable[[str], None] = None
-
-    def listen_for_user_input(self) -> None:
-        """Listens for the user input.
+    def __init__(self, agent_class: Type[Agent], user_id: str = None) -> None:
```
Comment (NoB0) [inline dialoguekit/platforms/terminal_platform.py:15]:
```suggestion
    def __init__(self, agent_class: Type[Agent], user_id: str = "terminal_user") -> None:
```

File: dialoguekit/platforms/terminal_platform.py [hunk 4]
```diff
@@ -3,37 +3,48 @@
...
+            agent_class: The class of the agent.
+            user_id: User ID (default: terminal_user).
         """
-        text = input("Your response: ")
-        if self._user_callback:
-            self._user_callback(text)
-
-    def display_agent_utterance(self, utterance: Utterance) -> None:
+        super().__init__(agent_class)
+        self._user_id = user_id or "terminal_user"
```
Comment (NoB0) [inline dialoguekit/platforms/terminal_platform.py:23]:
```suggestion
        self._user_id = user_id
```

**Review by NoB0 — COMMENTED** [review]
I added a couple of questions and suggestions. 
Can you please add the missing arguments in docstrings to pass pydocstyle check?

File: dialoguekit/core/utterance.py [hunk 5]
```diff
@@ -1,9 +1,13 @@
 """Interface representing the basic unit of communication."""
 
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (IKostric) [reply dialoguekit/core/utterance.py:3]: It is to enable forward references. DialogueParticipant is not available when Utterance is being imported. Alternative is to write DialogueParticipant in quotes, i.e., `participant: "DialogueParticipant" = field(hash=True)`.

The behaviour this import enables will be default in future versions of Python.

File: dialoguekit/participant/participant.py [hunk 6]
```diff
@@ -36,6 +37,14 @@ def id(self):
         """Returns the participant id."""
         return self._id
 
+    def get_dialogue_connector(self) -> DialogueConnector:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply dialoguekit/participant/participant.py:40]: Done

File: dialoguekit/platforms/platform.py (hunk 1 above)
Comment (IKostric) [reply dialoguekit/platforms/platform.py:122]: Done

File: dialoguekit/platforms/terminal_platform.py (hunk 2 above)
Comment (IKostric) [reply dialoguekit/platforms/terminal_platform.py:20]: Done

File: dialoguekit/platforms/terminal_platform.py (hunk 3 above)
Comment (IKostric) [reply dialoguekit/platforms/terminal_platform.py:15]: Done

File: dialoguekit/core/utterance.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (IKostric) [reply dialoguekit/core/utterance.py:3]: https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations

[the remark this answers is not in the document]
Comment (NoB0) [reply dialoguekit/core/utterance.py:3]: Good to know thanks for the details.

**Review by NoB0 — APPROVED** [review]
LGTM. Please fix tests before merging to main. 
See error in workflow: `ImportError: cannot import name 'Utterance' from partially initialized module 'dialoguekit.core' (most likely due to a circular import) (/home/runner/work/DialogueKit/DialogueKit/dialoguekit/core/__init__.py)`

**Pull request merged**